internal/service/elasticache/replication_group.go

```diff
@@ -816,6 +840,16 @@ func resourceReplicationGroupUpdate(d *schema.ResourceData, meta interface{}) er
 		requestUpdate = true
 	}
 
+	if d.HasChange("ip_discovery") {
+		params.IpDiscovery = aws.String(d.Get("ip_discovery").(string))
+		requestUpdate = true
+	}
+
+	if d.HasChange("network_type") {
+		req.IpDiscovery = aws.String(d.Get("network_type").(string))
```

Comment:
use `params` instead of `req`

```suggestion
		req.IpDiscovery = aws.String(d.Get("network_type").(string))
```